### PR TITLE
fix(files): Bring back handling of failed files

### PR DESCRIPTION
**Pull request by susnux**
* Follow up (broken by) #45788 

## Summary
The fileid might be set to -1 in case of an error, this brings back proper handling of this case.

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [ ] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Comment by susnux** [comment]
Best with, but no blocker: https://github.com/nextcloud-libraries/nextcloud-files/pull/985

File: apps/files/src/components/FileEntry/FileEntryActions.vue [hunk 1]
```diff
@@ -152,7 +153,7 @@ export default defineComponent({
 
 		// Sorted actions that are enabled for this node
 		enabledActions() {
-			if (this.source.attributes.failed) {
+			if (this.source.status === NodeStatus.FAILED) {
```
Comment (Fenn-CS) [inline apps/files/src/components/FileEntry/FileEntryActions.vue:155]: We can't use `isFailedSource()` from the mixin here?

**Review by nfebe — APPROVED** [review]

File: apps/files/src/components/FileEntry/FileEntryActions.vue (hunk 1 above)
Comment (susnux) [reply apps/files/src/components/FileEntry/FileEntryActions.vue:155]: Not without passing it from the file entry as a prop to the actions.
But I do not think it makes much sense to do so as we can easily (this one line) get it from the source.

**Review by artonge — APPROVED** [review]

**Pull request merged**